File: Math & Geometry/2073. Time Needed to Buy Tickets/2073.cpp
#pragma GCC optimize('0fast','unroll-loops')

class Solution {
public:
    int timeRequiredToBuy(vector<int>& tickets, int k) {
        ios_base::sync_with_stdio(0);
        cin.tie(0);
        cout.tie(0);

        int res=0;
        for(int i=0; i<tickets.size(); ++i){
            if(i<=k){
                res+=min(tickets[i],tickets[k]);
            } else{
                res+=min(tickets[i],tickets[k]-1);
            }
        }
        return res;
    }
};

File: Math & Geometry/2073. Time Needed to Buy Tickets/2073.py
class Solution:
    def timeRequiredToBuy(self, tickets: List[int], k: int) -> int:
        res=0
        for i,v in enumerate(tickets):
            if i<=k:
                res+=min(tickets[i],tickets[k])
            else:
                res+=min(tickets[i],tickets[k]-1)
        return res
        